### PR TITLE
Dave/dls 6969/accept result code 4

### PR DESCRIPTION
**Pull request by lutnos**

### CODE REVIEW

**Review by johno1985 — COMMENTED** [review]
minor comment

File: app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala [hunk 1]
```diff
@@ -174,15 +174,10 @@ class HelpToSaveServiceImpl @Inject() (helpToSaveProxyConnector: HelpToSaveProxy
...
-              if (res.resultCode === 4) {
-                metrics.itmpEligibilityCheckErrorCounter.inc()
-                pagerDutyAlerting.alert("Received result code 4 from DES eligibility check")
-                Left(s"Received result code 4 from DES eligibility check interface. Result was '${res.result}'. Reason was '${res.reason}'")
-              } else {
-                logger.debug(s"Call to check eligibility successful, received 200 (OK) ${timeString(time)}", nino, additionalParams)
-                Right(res)
-              }
+            }, res ⇒ {
+              logger.debug(s"Call to check eligibility successful, received 200 (OK) ${timeString(time)}", nino, additionalParams)
```
[the remark this answers is not in the document]
Comment (lutnos) [reply app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala]: Yes, good idea, thanks, have reinstate that

File: app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala [hunk 2]
```diff
@@ -175,7 +175,11 @@ class HelpToSaveServiceImpl @Inject() (helpToSaveProxyConnector: HelpToSaveProxy
                 pagerDutyAlerting.alert("Could not parse JSON in eligibility check response")
                 Left(e)
             }, res => {
-              logger.debug(s"Call to check eligibility successful, received 200 (OK) ${timeString(time)}", nino, additionalParams)
+              if (res.resultCode === 4) {
+                metrics.itmpEligibilityCheckErrorCounter.inc()
```
Comment (johno1985) [inline app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala:179]: just wondering if we should log something/improve the logging here?

**Review by johno1985 — APPROVED** [review]
happy to approve, left a minor comment about logging though

File: app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (lutnos) [reply app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala:178]: Changed my mind! No it is no longer an error in our view, therefor should not be counted in the metric

File: app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala (hunk 2 above)
Comment (lutnos) [reply app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala]: I don't thing we need to, the response is already logged at INFO by the statement before the match at line 168

**Review by johno1985 — DISMISSED** [review]

**Review by TomRafferty — APPROVED** [review]

**Pull request merged**